src/util.h

```diff
@@ -29,6 +29,15 @@
 #include <boost/signals2/signal.hpp>
 #include <boost/thread/exceptions.hpp>
 
+#ifdef DEBUG_ASSERTION
+/// If DEBUG_ASSERTION is enabled this asserts when the predicate is false.
+//  If DEBUG_ASSERTION is disabled and the predicate is false, it executes the execInRelease statements.
+//  Typically, the programmer will error out -- return false, raise an exception, etc in the execInRelease code.
+#define DbgAssert(pred, execInRelease) do { assert(pred); } while(0)
+#else
+#define DbgAssert(pred, execInRelease) do { if (!(pred)) { execInRelease; }} while(0)
```

Comment:
@dgenr8 I was concerned about that.  But dropping the do {} while() has its own problems around hidden "ifs".  Do you have a suggestion?